src/base.js

```diff
@@ -375,7 +376,7 @@ function getObject(tableName, subject) {
                     cache.cacheData(tableName+"::"+subject, item, options.cacheSeconds);
                     resolve(item);
                 }
-                resolve(null);
+                reject("Item does not exist: " + tableName + "::" + subject);
```

Comment:
@optimuspaul Thoughts on throwing errors vs just returning null. It might not makes sense to do this in this PR regardless, as we want to be consistent across the entire project.